pkg/chunk/cache/fifo_cache_test.go

```diff
@@ -2,100 +2,161 @@ package cache
+	assert.Equal(t, testutil.ToFloat64(c.entriesEvicted), float64(overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.entriesCurrent), float64(cnt))
+	assert.Equal(t, testutil.ToFloat64(c.totalGets), float64(cnt))
+	assert.Equal(t, testutil.ToFloat64(c.totalMisses), float64(0))
+	assert.Equal(t, testutil.ToFloat64(c.staleGets), float64(0))
+	assert.Equal(t, testutil.ToFloat64(c.memoryBytes), float64(cnt*sizeOf(itemTemplate)))
 
-	for i := 0; i < size-overwrite; i++ {
-		_, ok := c.Get(ctx, strconv.Itoa(i))
+	for i := 0; i < cnt-overwrite; i++ {
+		_, ok := c.Get(ctx, fmt.Sprintf("%02d", i))
 		require.False(t, ok)
 	}
-	for i := size; i < size+overwrite; i++ {
-		value, ok := c.Get(ctx, strconv.Itoa(i))
+	for i := cnt; i < cnt+overwrite; i++ {
+		value, ok := c.Get(ctx, fmt.Sprintf("%02d", i))
 		require.True(t, ok)
 		require.Equal(t, i, value.(int))
 	}
 
+	assert.Equal(t, testutil.ToFloat64(c.entriesAdded), float64(2))
+	assert.Equal(t, testutil.ToFloat64(c.entriesAddedNew), float64(cnt+overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.entriesEvicted), float64(overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.entriesCurrent), float64(cnt))
+	assert.Equal(t, testutil.ToFloat64(c.totalGets), float64(cnt*2))
+	assert.Equal(t, testutil.ToFloat64(c.totalMisses), float64(cnt-overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.staleGets), float64(0))
+	assert.Equal(t, testutil.ToFloat64(c.memoryBytes), float64(cnt*sizeOf(itemTemplate)))
+
 	// Check updates work
 	keys = []string{}
 	values = []interface{}{}
-	for i := size; i < size+overwrite; i++ {
-		keys = append(keys, strconv.Itoa(i))
+	for i := cnt; i < cnt+overwrite; i++ {
+		keys = append(keys, fmt.Sprintf("%02d", i))
 		values = append(values, i*2)
 	}
 	c.Put(ctx, keys, values)
-	require.Len(t, c.index, size)
-	require.Len(t, c.entries, size)
+	require.Len(t, c.entries, cnt)
 
-	for i := size; i < size+overwrite; i++ {
-		value, ok := c.Get(ctx, strconv.Itoa(i))
+	for i := cnt; i < cnt+overwrite; i++ {
+		value, ok := c.Get(ctx, fmt.Sprintf("%02d", i))
 		require.True(t, ok)
 		require.Equal(t, i*2, value.(int))
 	}
+
+	assert.Equal(t, testutil.ToFloat64(c.entriesAdded), float64(3))
+	assert.Equal(t, testutil.ToFloat64(c.entriesAddedNew), float64(cnt+overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.entriesEvicted), float64(overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.entriesCurrent), float64(cnt))
+	assert.Equal(t, testutil.ToFloat64(c.totalGets), float64(cnt*2+overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.totalMisses), float64(cnt-overwrite))
+	assert.Equal(t, testutil.ToFloat64(c.staleGets), float64(0))
+	assert.Equal(t, testutil.ToFloat64(c.memoryBytes), float64(cnt*sizeOf(itemTemplate)))
+
+	c.Stop()
 }
 
-func TestFifoCacheEvictionExpiry(t *testing.T) {
-	c := NewFifoCache("test2", FifoCacheConfig{Size: 3, Validity: 5 * time.Millisecond})
-	ctx := context.Background()
-	memorySz := int(unsafe.Sizeof(cacheEntry{})) * 3
+func TestFifoCacheExpiry(t *testing.T) {
```

Comment:
If you change this test to the table pattern you will not have to pass `keys` and `data` to the other function.